pandas/indexes/numeric.py

```diff
@@ -334,7 +312,7 @@ def equals(self, other):
                 return False
             left, right = self._values, other._values
             return ((left == right) | (self._isnan & other._isnan)).all()
-        except TypeError:
+        except (TypeError, ValueError):
```

Comment:
```
In [2]: pd.Int64Index([1,2,3]).equals(Series([1,2,3]))
Out[2]: True
```

yes see #13107 this should be ``False``

cc @sinhrks (maybe we should fix that first), don't want this PR to keep expanding. Alternatively could add a TODO and come back later / ignore for now.